src/Algebra/Graph/Acyclic/Ord.hs

```diff
@@ -0,0 +1,164 @@
+-----------------------------------------------------------------
+-- |
+-- Module     : Algebra.Graph..Acyclic.Ord
+-- License    : MIT (see the file LICENSE)
+-- Stability  : experimental
+--
+-- __Alga__ is a library for algebraic construction and manipulation of graphs
+-- in Haskell. See <https://github.com/snowleopard/alga-paper this paper> for the
+-- motivation behind the library, the underlying theory, and implementation details.
+--
+-- This module defines functions which guarantee Acyclic graph by
+-- establishing an invariant on the edges.
+-- An edge (x, y) exists if and only if x < y. This way no back
+-- edges are formed and hence making an acyclic graph is impossible.
```

Comment:
I think you meant "hence making a cyclic graph is impossible" :)

Reply:
Ah! yes, my bad.